app/src/main/java/com/parasoft/demoapp/HomeActivity.java

```diff
@@ -136,17 +153,61 @@ public void openOrderDialog(String orderNumber) {
         orderDialog.show(getSupportFragmentManager(), OrderDialog.TAG);
     }
 
-    public void initRecyclerView(List<OrderResponse> orders) {
+    public void initRecyclerView() {
+        page = 0;
+        getData(page);
         LinearLayoutManager layoutManager = new LinearLayoutManager(this);
         recyclerView.setLayoutManager(layoutManager);
-        OrderAdapter orderAdapter = new OrderAdapter(orders, item -> {
+        orderAdapter = new OrderAdapter(orderDisplayList, item -> {
             if (orderItemClickable) {
                 openOrderDialog(item.getOrderNumber());
             }
         });
         recyclerView.setAdapter(orderAdapter);
         recyclerView.setVisibility(View.VISIBLE);
-        orderAdapter.setCanStart(true);
+
+        initListener();
+    }
+
+    private void initListener() {
+        // slide-up load more
+        recyclerView.addOnScrollListener(new RecycleOnScrollListener() {
+            @Override
+            public void onLoadMore() {
+                // to avoid possible repeated slide-up operation
+                if (OrderAdapter.LOADING == orderAdapter.getLoadState()) {
+                    orderAdapter.notifyItemRemoved(orderAdapter.getItemCount());
+                    return;
+                }
+
+                orderAdapter.setLoadState(OrderAdapter.LOADING);
+                if (orderDisplayList.size() < orderResponseList.size()) {
+                    new Timer().schedule(new TimerTask() {
+                        @Override
+                        public void run() {
+                            runOnUiThread(() -> {
+                                page++;
+                                getData(page);
+                                orderAdapter.setLoadState(OrderAdapter.LOAD_FINISH);
+                            });
+                        }
+                    }, 3000);
+                } else {
+                    page = 0;
+                    orderAdapter.setLoadState(OrderAdapter.LOAD_END);
+                }
+            }
+        });
+        // slide-down refresh
+        ordersLoader.setOnRefreshListener(() -> new Handler().postDelayed(() -> loadOrderList(false), 3000));
```

Comment:
Same for the below question, I was thinking about it at the time only init RecycleView when neccessary instead of intializing every time we fresh. Thanks for the fix.